marconi-chain-index/src/Marconi/ChainIndex/CLI.hs

```diff
@@ -42,38 +42,82 @@ import Marconi.ChainIndex.Types (
  )
 import Paths_marconi_chain_index (version)
 
-{- | Allow the user to set a starting point for indexing the user needs to provide both
- a @BlockHeaderHash@ (encoded in RawBytesHex) and a @SlotNo@ (a natural number).
+-- | Represents a specified point from which to start indexing
+data StartFrom
+  = StartFromGenesis
```

Comment:
I'm not 100% sure here, but I'm wondering if we start from genesis, should we delete the marconi database files? In the case where resuming is enabled, the indexer will not reinitialise it's database. Instead, it will drain incoming event if it has already indexed them.

Probably a new ticket for that, but worth discussing it here.

cc @berewt

Reply:
I don't think it should just delete with this flag, I think another flag should be required for that. It's a potential nasty sharp edge for users otherwise; someone might just want to try out the command without expecting it to wipe their DB.

Reply:
You may want to start again from a given point without wiping out the database. It's not obvious for genesis, but as it would be reasonable for other points, having the same behaviour for genesis is consistent.

Reply:
Good points. Also, having an additional option such as `--delete-database` makes sense. That should be part of a separate PR.